ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/FilterStringColLikeStringScalar.java

```diff
@@ -140,4 +133,93 @@ public Checker tryCreate(String pattern) {
       return new ComplexChecker("^" + UDFLike.likePatternToRegExp(pattern) + "$");
     }
   }
+
+  private enum PatternType {
+    NONE, // "abc"
+    BEGIN, // "abc%"
+    END, // "%abc"
+    MIDDLE, // "%abc%"
+    CHAINED, // "abc%def%ghi%"
+    COMPLEX, // all other cases, such as "ab%c_de"
+  }
+
+  /**
+   * Check the LIKE pattern..
+   */
+  private static class CheckerPattern {
+    private final String pattern;
+    private final PatternType type;
+
+    public CheckerPattern(PatternType type, String pattern) {
+      this.type = type;
+      this.pattern = pattern;
+    }
+
+    public String check() {
+      PatternType lastType = PatternType.NONE;
+      int length = pattern.length();
+      int beginIndex = 0;
+      int endIndex = length;
+      char lastChar = 'a';
```

Comment:
it doesn't seem to be a generic code. why do we hardcode here?

Reply:
This is a value for initialization. There is no problem if it is a value other than `\`, so Initialize it to 0 for clarity.